Runtime/Highlight/ObjectHighlighter.cs

```diff
@@ -0,0 +1,453 @@
+﻿using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Innoactive.Creator.Unity;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    /// <summary>
+    /// Adds highlighting functionality to a GameObject with Renderers.
+    /// </summary>
+    /// <remarks>
+    /// Highlights are always queued following a LIFO (Last In First Out) scheme. 
+    /// </remarks>
+    public class ObjectHighlighter : MonoBehaviour
+    {
+        private readonly Dictionary<string, Material> highlightIDs = new Dictionary<string, Material>();
+
+        protected bool IsBeingHighlighted => highlightIDs.Any();
+
+        [SerializeField, HideInInspector]
+        private Renderer[] renderers = {};
+        
+        [SerializeField, HideInInspector]
+        private MeshRenderer highlightMeshRenderer = null;
+        
+        [SerializeField , HideInInspector]
+        private MeshFilter highlightMeshFilter = null;
+        
+        protected virtual void Reset()
+        {
+            RefreshCachedRenderers();
+        }
+
+        protected virtual void OnDisable()
+        {
+            if (IsBeingHighlighted)
+            {
+                ReenableRenderers();
+                highlightIDs.Clear();
+            }
+        }
+
+        /// <summary>
+        /// Returns true if there is any highlight active.
+        /// </summary>
+        /// <param name="listOfActiveHighlights">In case there is any active highlight, this value will be filled with all available highlight IDs.</param>
+        public bool IsHighlighting(out IEnumerable<string> listOfActiveHighlights)
```

Comment:
Why do we have the listOfActiveHighlights here? you are combining two different logical methods in one.

Reply:
A highlighter can highlight multiple times the same object, only the last requested highlight is shown, this method intends to tell if there is any highlight going on and to provide a list of which highlights are happening at the same time.

I can split it if you think is cleaner that way.